### PR TITLE
feat: mirror workflow images

### PR DESCRIPTION
**Pull request by alexcodelf**
**Problem:**

**Solution:**
Support mirror workflow images, modify step images, support image repository for workflows.
**Related Issue:**
#412

### CODE REVIEW

File: hack/mirror/mirror_image_list.txt [hunk 1]
```diff
@@ -4,3 +4,6 @@ quay.io/prometheus-operator/prometheus-config-reloader:v0.67.0  docker.io/sealio
 quay.io/prometheus/prometheus:v2.46.0                           docker.io/sealio/mirrored-prometheus:v2.46.0
 registry.k8s.io/kube-state-metrics/kube-state-metrics:v2.9.2    docker.io/sealio/mirrored-kube-state-metrics:v2.9.2
 quay.io/kubecost1/kubecost-cost-model:1.105.2                   docker.io/sealio/mirrored-kubecost-cost-model:v1.105.2
+quay.io/argoproj/workflow-controller:v3.5.0                     docker.io/sealio/mirrored-workflow-controller:v3.5.0
```
Comment (aiwantaozi) [inline hack/mirror/mirror_image_list.txt:7]: do we need to mirror badouralix/curl-jq

File: pkg/settings/settings.go [hunk 2]
```diff
@@ -56,8 +56,7 @@ var (
 		"WorkflowStepServiceImage",
 		editable,
 		// When the image is updated, sync the one in server Dockerfile.
-		// TODO replace with seal curl-jq image.
-		initializeFrom("badouralix/curl-jq"),
+		initializeFrom("sealio/curl-jq:v0.0.1"),
```
Comment (aiwantaozi) [inline pkg/settings/settings.go:59]: may update after https://github.com/seal-io/walrus/pull/1472/files

File: hack/mirror/mirror_image_list.txt (hunk 1 above)
Comment (alexcodelf) [reply hack/mirror/mirror_image_list.txt:7]: The image is replaced by sealio/curl-jq

**Review by aiwantaozi — APPROVED** [review]
LGTM

**Pull request merged**